packages/frontend/src/app/utils.ts

```diff
@@ -44,3 +49,34 @@ export const createQueryString = (
 export const isValidHex = (hex: string) => {
   return Boolean(hex.match(/^0x[0-9a-f]+$/i));
 };
+export const getChainById = (chainId: number) => {
+  const chains = useChains();
+  return chains.find((chain) => chainId === chain.id);
+};
+export const getPublicClient = (chainId: number) => {
+  return createPublicClient({
+    chain: getChainById(chainId),
+    transport: http(),
+  });
+};
+export const getTokenInformation = async (
+  publicClient: PublicClient,
+  tokenAddress: `0x${string}`,
+) => {
+  if (tokenAddress === zeroAddress) {
+    return { symbol: "ETH", decimal: 18 };
+  }
+  const symbol = (await publicClient.readContract({
+    address: tokenAddress,
+    abi: abi.ERC20,
+    functionName: "symbol",
+    args: [],
+  })) as string;
+  const decimal = (await publicClient.readContract({
+    address: tokenAddress,
+    abi: abi.ERC20,
+    functionName: "decimals",
+    args: [],
+  })) as number;
+  return { symbol, decimal };
```

Comment:
reading the contract for the Symbol and the Decimal should done concurrently. Collect the promises from both operations then use `Promise.all` to resolve